core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -124,4 +176,81 @@ class ReplicaFetcherThread(name:String,
   def handlePartitionsWithErrors(partitions: Iterable[TopicAndPartition]) {
     delayPartitions(partitions, brokerConfig.replicaFetchBackoffMs.toLong)
   }
+
+  protected def fetch(fetchRequest: FetchRequest): Map[TopicAndPartition, PartitionData] = {
+    val clientResponse = sendRequest(ApiKeys.FETCH, Some(fetchRequestVersion), fetchRequest.underlying)
+    new FetchResponse(clientResponse.responseBody).responseData.asScala.map { case (key, value) =>
+      TopicAndPartition(key.topic, key.partition) -> new PartitionData(value)
+    }
+  }
+
+  private def sendRequest(apiKey: ApiKeys, apiVersion: Option[Short], request: AbstractRequest): ClientResponse = {
```

Comment:
We don't need to replicate the reconnect behavior in SimpleConsumer. However, we probably need to close the client connection on socketTimeoutException.